### DISCUSSION TITLE
Routes outside of `docs/` with sidebar with docusaurus v2

### DISCUSSION THREAD

**Discussion by JamesSingleton** [facebook/docusaurus#4179] [Q&A]
So either I am missing it in the documentation of it doesn't exist. I know that if you put everything under `docs/`, and the top of your markdown file looks like
```md
---
id: cliCommands
title: CLI Commands
sidebar_label: CLI Commands
slug: /server/cli-commands
---
```
You can go to a route like `docs/server/cli-commands` and it works. However, I wanted to go to a URL like `guides/adding-styles`. I have created a `guides/` folder at the same level as the `docs/` and I added the following to my docusaurus.config.js
```js
{
  to: 'guides',
  label: 'Guides',
  position: 'left',
},
```

And the following in my `adding-styles` markdown file under `guides/`
```md
---
id: addingStyles
title: Adding Styles
sidebar_label: Adding Styles
slug: /adding-styles
---
```

However, it seems that I am unable to go to `guides/adding-styles`

**Comment by slorber** [comment]
I have a hard time understanding the final FS structure / config you have.

Can you use http://new.docusaurus.io/ and create a small repro? 
And give me full URLs (which doc should live at which URL?)

Using relative paths like `guides/styles` is quite abstract, as relative paths are resolved depending on the current URL, not the root of your site, so  the same relative path can lead to 2 different final urls

**Reply by JamesSingleton** [reply]
@slorber sorry for the delay, here is what I mean https://codesandbox.io/s/ecstatic-drake-clz3n

**Reply by slorber** [reply]
@JamesSingleton it's only half of what I asked for 😅 

> And give me full URLs (which doc should live at which URL?)

I have a codesandbox but I don't know what I should look for in it, and what you get/expect. It just works as intended.

I have absolutely 0 idea of how your sandbox is different from the initial sandbox, and what you modified here

**Reply by JamesSingleton** [reply]
If you click on guides in the navbar it doesn't take you to the expected page. Given the slug in the markdown and the path given in the config file, I would expect it to.

**Reply by JamesSingleton** [reply]
@slorber do if I go to https://clz3n.sse.codesandbox.io/guides/styles on that sandbox, it does not show as I would expect.

**Reply by slorber** [reply]
@JamesSingleton that's a false expectation. We don't read md files outside of the ./docs folder, as explained in the doc. 
The path in the config is just a URL path used as a link. It does not know anything about markdown files.

If you want:
- you can configure the docs plugin path to read md files from `./guides` instead of `./docs`
- you can configure `routeBasePath: '/'` or `routeBasePath: '/guides/'` to the docs plugin if you don't like to have the `/docs/` path prefix added to all docs (ie you would be able to use `slug: /guides/xyz`)
- you can use multi-instance of docs: https://v2.docusaurus.io/docs/next/docs-multi-instance

We offer quite a few flexible options that you'll have to choose according to your usecase

Just remember that adding navbar links does not create any new page nor trigger reading md files. The pages should exist in the first place, and then you link to them using the theme navbar config (using URL paths, not md paths)
